contracts/libraries/AccountLib.sol

```diff
@@ -217,13 +215,16 @@ function cache(AccountSystem storage self, address token, bool status) {
 
 /**
  * @dev Computes surplus (positive) or deficit (negative) in actual tokens compared to tracked amounts.
+ * @return net Net balance of physical - virtual tokens in native token decimals.
  */
 function getNetBalance(
     AccountSystem storage self,
     address token,
     address account
 ) view returns (int256 net) {
-    uint256 internalBalance = self.reserves[token];
+    uint256 internalBalanceWad = self.reserves[token];
+    uint256 internalBalance =
+        AssemblyLib.scaleFromWadUp(internalBalanceWad, IERC20(token).decimals());
```

Comment:
The `reserves` now track all token amounts in WAD units, even if the token has less than 18 decimals. The `balanceOf` function returns native token decimals. 

Consequence: `reserves` is scaled FROM WAD  and ROUNDED UP to always align the reserves and physical balance in a way that transfers more tokens into the contract than there is in a fractional amount.

E.g.: The reserves could have 1003.53 tokens, which is then rounded up to 1004, so 1004 physical tokens are transferred in, even though the reserves have less. This maintains the CRITICAL invariant that `physical balance => virtual reserve`.